vscode/src/completions/utils/lexical-analysis.ts

```diff
@@ -0,0 +1,161 @@
+import path from 'path'
+
+import Parser, { Point, SyntaxNode, Tree } from 'web-tree-sitter'
+
+/**
+ * List of all supported languages that we have grammars and
+ * lexems for. Note that enum values are copied from VSCode API,
+ * if we want to make it work with different editors we should
+ * enhance language detection.
+ *
+ * TODO: Decouple language detect to make it editor agnostic
+ */
+export enum SupportedLanguage {
+    JavaScript = 'javascript',
+    TypeScript = 'typescript',
+    Java = 'java',
+    Go = 'go',
+    // The problem with these two is that we have to use typescript
+    // wasm module grammar, the problem that exports in wasm works differently
+    // we probably have to compile custom wasm module for tsx grammar.
+    // JSX = 'javascriptreact',
+    // TSX = 'typescriptreact',
+}
```

Comment:
What do you think of using a literal string enum instead? 

```suggestion
export type SupportedLanguage =  'javascript' | 'typescript' | 'java' | 'go'
```

Then below we don't have to write long assignments like `[SupportedLanguage.JavaScript]` but can just use `javascript` instead 😬

Reply:
I don't mind, but I like enum values more because with them, we could avoid editing strings in multiple places if we have to change them later. It's a common problem that I've seen in the main web app.

Reply:
Makes sense. I think these strings are pretty stable though, VS Code uses them as their language ID so I doubt they'll change them